pallets/allocations/src/lib.rs

```diff
@@ -102,6 +171,30 @@ pub mod pallet {
 	#[pallet::generate_store(pub(super) trait Store)]
 	pub struct Pallet<T>(PhantomData<T>);
 
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn on_initialize(n: BlockNumberFor<T>) -> Weight {
```

Comment:
Shall we do like the grants/vesting pallet and use blocks from the relay chain which might be slightly more stable?

Reply:
@ETeissonniere Switching to using the relay chain block number comes with its cons and pros. The pros as you mentioned is that the timing would be more stable. There are however a few cons. First is we couple an action on the parachain to an event on the relay chain which is first an additional reliance and coupling to the relay chain. Second you'd always need to remember to find out and use the block number from the relay chain when you need to verify or predict an action. I think we decided to pay this price for our grants because they are based on or resemble financial contracts so they'd better be more predictable time wise. However for setting an upper bound (especially that we are not going to go for a very tight bound), the simplicity and independence might be better.  What do you think?

Reply:
The independence argument is certainly pretty strong. Ultimately it comes down to the guarantees we seek to provide. Do we want a very stable measure of time or are we fine with tolerating drifts of a few seconds for some blocks (which could also accumulate over time)? From what you are saying, it sounds like we could go with the former. Correct?

Reply:
Correct, implementation wise we can go with the relay chain block pace for timing and it should be a pretty trivial change. I was on the side that a few seconds drifts and its accumulating effect for an upper bound could be okay. Especially that we know it's not going to to make intervals shorter. Parachain days will be only longer than the actual day. But this would also mean the widening or tightening of the funnel would potentially take longer than an actual quarter to happen (unless block production on parachain remains more stable going forward). I'm open to make the change and use the relay chain block pace, what would you suggest?

Reply:
For the record here, the implementation is changed to use the relay chain block numbers instead.